tests/blockchain/eth/entities/actors/test_stakeholder.py

```diff
@@ -0,0 +1,219 @@
+    assert first_config_contents == second_config_contents
+
+
+def test_initialize_stake_with_existing_staking_account(staking_software_stakeholder, stake_value, token_economics):
+
+    stake = staking_software_stakeholder.stakes[0]
+
+    # Stake, deriving a new account, using tokens and ethers from the funding account
+    stake = staking_software_stakeholder.initialize_stake(checksum_address=stake.owner_address,
+                                                          amount=stake_value,
+                                                          duration=token_economics.minimum_locked_periods)
+
+    assert stake.blockchain == staking_software_stakeholder.blockchain
+    assert stake.value == stake_value
+    assert stake.duration == token_economics.minimum_locked_periods
+    assert stake.owner_address != staking_software_stakeholder.funding_account
+
+    staking_software_stakeholder.blockchain.time_travel(periods=1)  # Wait for stake to begin
+    staking_agent = Agency.get_agent(StakingEscrowAgent)
+    stakes = list(staking_agent.get_all_stakes(staker_address=stake.owner_address))
+    assert len(stakes) == 2
+
+
+def test_initialize_stake_with_new_account(staking_software_stakeholder, stake_value, token_economics):
+
+    # Stake, deriving a new account, using tokens and ethers from the funding account
+    stake = staking_software_stakeholder.initialize_stake(password=INSECURE_DEVELOPMENT_PASSWORD,
+                                                          amount=stake_value,
+                                                          duration=token_economics.minimum_locked_periods)
+
+    assert stake.blockchain == staking_software_stakeholder.blockchain
+    assert stake.value == stake_value
+    assert stake.duration == token_economics.minimum_locked_periods
+    assert stake.owner_address != staking_software_stakeholder.funding_account
+
+    staking_agent = Agency.get_agent(StakingEscrowAgent)
+    stakes = list(staking_agent.get_all_stakes(staker_address=stake.owner_address))
+    assert len(stakes) == 1
+
+
+def test_divide_stake(staking_software_stakeholder, token_economics):
+    stake = staking_software_stakeholder.stakes[1]
+
+    target_value = token_economics.minimum_allowed_locked
+    original_stake, new_stake = staking_software_stakeholder.divide_stake(address=stake.owner_address,
+                                                                          password=INSECURE_DEVELOPMENT_PASSWORD,
+                                                                          index=0,
+                                                                          duration=10,
+                                                                          value=target_value)
+
+    staking_agent = Agency.get_agent(StakingEscrowAgent)
+    stakes = list(staking_agent.get_all_stakes(staker_address=stake.owner_address))
+    assert len(stakes) == 2
+    assert new_stake.value == target_value
```

Comment:
Is it possible to also check the value of `original_stake` after the divide. Something akin to (?):
```python
total_stake_value = staking_software_stakeholder.total_stake()
# ... do the divide
assert original_stake.value == (total_stake_value - target_value)
```

Reply:
:heavy_check_mark: